src/Polly/RateLimit/AsyncRateLimitPolicy.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Polly.RateLimit
+{
+    /// <summary>
+    /// A rate-limit policy that can be applied to asynchronous delegates.
+    /// </summary>
+    public class AsyncRateLimitPolicy : AsyncPolicy, IRateLimitPolicy
+    {
+        private readonly IRateLimiter _rateLimiter;
+
+        internal AsyncRateLimitPolicy(IRateLimiter rateLimiter)
+        {
+            _rateLimiter = rateLimiter ?? throw new NullReferenceException(nameof(rateLimiter));
```

Comment:
Should these be this instead?

```suggestion
            _rateLimiter = rateLimiter ?? throw new ArgumentNullException(nameof(rateLimiter));
```